packages/app-headless-cms/src/admin/views/contentEntries/Table/Main.tsx

```diff
@@ -1,66 +1,39 @@
-import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import React, { useCallback, useEffect, useRef, useState } from "react";
 import debounce from "lodash/debounce";
-import { FolderDialogCreate, useAcoList } from "@webiny/app-aco";
+import { FolderDialogCreate } from "@webiny/app-aco";
 import { Scrollbar } from "@webiny/ui/Scrollbar";
 import { Empty } from "~/admin/components/ContentEntries/Empty";
 import { Header } from "~/admin/components/ContentEntries/Header";
 import { LoadingMore } from "~/admin/components/ContentEntries/LoadingMore";
 import { LoadMoreButton } from "~/admin/components/ContentEntries/LoadMoreButton";
 import { Table } from "~/admin/components/ContentEntries/Table";
 import { MainContainer, Wrapper } from "./styled";
-import {
-    ListMeta,
-    ListSearchRecordsSort,
-    ListSearchRecordsSortItem,
-    ListSearchRecordsWhereQueryVariables
-} from "@webiny/app-aco/types";
-import { Sorting } from "@webiny/ui/DataTable";
-import { useContentEntry } from "~/admin/views/contentEntries/hooks";
+import { useContentEntriesList, useContentEntry } from "~/admin/views/contentEntries/hooks";
 import { ContentEntry } from "~/admin/views/contentEntries/ContentEntry";
 import { useRouter } from "@webiny/react-router";
-import { CmsContentEntry } from "@webiny/app-headless-cms-common/types";
-import {
-    transformCmsContentEntriesToRecordEntries,
-    transformFolderItemsToFolderEntries
-} from "~/utils/acoRecordTransform";
+
 import { FOLDER_ID_DEFAULT } from "~/admin/constants";
 
 interface Props {
     folderId?: string;
 }
 
-const createSort = (sorting?: Sorting): ListSearchRecordsSort | undefined => {
-    if (!sorting?.length) {
-        return undefined;
-    }
-    return sorting.reduce<ListSearchRecordsSort>((items, item) => {
-        const sort = `${item.id}_${item.desc ? "DESC" : "ASC"}` as ListSearchRecordsSortItem;
-        if (items.includes(sort)) {
-            return items;
-        }
-        items.push(sort);
-        return items;
-    }, []);
-};
-
 export const Main: React.VFC<Props> = ({ folderId: initialFolderId }) => {
     const folderId = initialFolderId === undefined ? FOLDER_ID_DEFAULT : initialFolderId;
     const {
-        /**
-         * TODO refactor useAcoList to accept exact generic type
-         * We know that records are CmsContentEntry[] so we can safely cast afterwards
-         */
-        records: initialRecords,
-        folders: initialFolders,
+        records,
+        folders,
         listTitle,
         meta,
         isListLoading,
         isListLoadingMore,
-        listItems,
-        limit
-    } = useAcoList({
-        folderId: initialFolderId || FOLDER_ID_DEFAULT
-    });
+        isSearch,
+        search,
+        setSearch,
+        listMoreRecords,
+        sorting,
+        setSorting
+    } = useContentEntriesList({ folderId });
```

Comment:
I would recommend _not_ to destructure this, but assign it to a variable, like this:

```
const list = useContentEntriesList({ folderId });
```

I did this in File Manager, and it makes reading the component a lot easier, you know exactly if it's a local state/callback, or something coming from a hook. Moreover, you won't need to modify this destructuring when you add new properties, they'll simply be part of the `list` object.